etna/datasets/tsdataset.py

```diff
@@ -230,6 +249,22 @@ def __getitem__(self, item):
         df = df.loc[first_valid_idx:]
         return df
 
+    @staticmethod
+    def _expand_index(df: pd.DataFrame, freq: Optional[str], future_steps: int) -> pd.DataFrame:
+        if freq is None:
+            new_index = np.arange(df.index.min(), df.index.max() + future_steps + 1)
+        else:
+            with warnings.catch_warnings():
+                warnings.filterwarnings(action="ignore", message="Argument `closed` is deprecated")
+                future_dates = pd.date_range(start=df.index.max(), periods=future_steps + 1, freq=freq, closed="right")
+            new_index = df.index.append(future_dates)
```

Comment:
Why do we generate `new_index` for int "timestamp" from `min` to `max + future_steps +1`, when for date "timestamp" we generate from `max` to `max + future_steps + 1` and then append to the original index? Is there a hidden logic here?

Reply:
For int timestamp it is easy to make `arange`.

For datetime timestamp I just keep current code.